src/main/scala/io/iohk/ethereum/network/PeerEventBusActor.scala

```diff
@@ -0,0 +1,131 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.event.ActorEventBus
+import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.{MessageFromPeer, MessageToPeer, PeerDisconnected, PeerHandshakeSuccessful}
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier._
+import io.iohk.ethereum.network.handshaker.Handshaker.HandshakeResult
+import io.iohk.ethereum.network.p2p.Message
+
+object PeerEventBusActor {
+  def props: Props = Props(new PeerEventBusActor)
+
+  sealed trait PeerSelector {
+    def contains(peerId: PeerId): Boolean
+  }
+
+  object PeerSelector {
+
+    case object AllPeers extends PeerSelector {
+      override def contains(p: PeerId): Boolean = true
+    }
+
+    case class WithId(peerId: PeerId) extends PeerSelector {
+      override def contains(p: PeerId): Boolean = p == peerId
+    }
+
+  }
+
+  sealed trait SubscriptionClassifier
+
+  object SubscriptionClassifier {
+    case class MessageReceivedClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
+    case class MessageSentClassifier(messageCodes: Set[Int], peerSelector: PeerSelector) extends SubscriptionClassifier
```

Comment:
It could be done that way, but what I prefer about this implementation is that there's no need for each time a message is sent (that is required by `PeersInfoHolderActor` or any future actor needing it) also sending the update to it (which we might forget).
Also regarding `PeersInfoHolderActor` being the one that forwards the message, at least the intent of the `PeersInfoHolder` was being a passive entity as much as possible, but we could rename the actor so as to also handle doing that.